File: src/api/assignment-remove-file.ts
import { RequestFN } from "~/core/request-function";
import { EntityState, TabLocation, type SessionHandle } from "~/models";

export const assignmentRemoveFile = async (session: SessionHandle, assignmentID: string): Promise<void> => {
  const request = new RequestFN(session, "SaisieTAFARendreEleve", {
    _Signature_: { onglet: TabLocation.Assignments },

    donnees: {
      listeFichiers: [{
        E: EntityState.DELETION,
        TAF: { N: assignmentID }
      }]
    }
  });

  await request.send();
};

File: library/src/commonMain/kotlin/api/assignment-remove-file.kt
package api

import core.RequestFN
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.*
import models.EntityState
import models.SessionHandle
import models.TabLocation

suspend fun assignmentRemoveFile (session: SessionHandle, assignmentID: String) {
    val request = RequestFN(session.information, "SaisieTAFARendreEleve", Json.encodeToString(buildJsonObject {
        putJsonObject("_Signature_") {
            put("onglet", TabLocation.Assignments.code)
        }

        putJsonObject("donnees") {
            putJsonArray("listeFichiers") {
                addJsonObject {
                    put("E", EntityState.DELETION.code)
                    putJsonObject("TAF") {
                        put("N", assignmentID)
                    }
                }
            }
        }
    }))

    request.send()
}